time-series/time-series-api/src/main/java/com/powsybl/timeseries/ast/TimeSeriesNames.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2018, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.timeseries.ast;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class TimeSeriesNames extends DefaultNodeCalcVisitor<Void, Set<String>> {
+
+    public static Set<String> list(NodeCalc nodeCalc) {
+        Objects.requireNonNull(nodeCalc);
+        Set<String> timeSeriesNames = new TreeSet<>();
+        nodeCalc.accept(new TimeSeriesNames(), timeSeriesNames);
+        return timeSeriesNames;
+    }
+
+    @Override
+    public Void visit(TimeSeriesNameNodeCalc nodeCalc, Set<String> timeSeriesNames) {
```

Comment:
Assert that nodeCalc and timeSeriesNames are not null?

Reply:
It cannot be null